### PR TITLE
Upgrade Core Addons to support Default Versions from AWS sdk for different Kubernetes versions 

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*
Upgrade Core Addons to support MAP of Versions for Kubernetes versions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/builders/observability-builder.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export class ObservabilityBuilder extends BlueprintBuilder {
     private certManagerProps: addons.CertManagerAddOnProps;
     private cloudWatchInsightsAddOnProps: addons.CloudWatchInsightsAddOnProps;
     private coreDnsProps: addons.CoreDnsAddOnProps;
+    private CoreDnsVersion: string = "auto";
```
Comment (shapirov103) [inline lib/builders/observability-builder.ts:15]: variables are lower-camel case.

**Review by shapirov103 — COMMENTED** [review]
one minor comment. 
we need to think about how to maintain these versions across releases. For example, step 1 could be making an API call at synth time and checking wether the version used is indeed the compatible/default version for the target EKS version.

File: lib/builders/observability-builder.ts (hunk 1 above)
Comment (elamaran11) [reply lib/builders/observability-builder.ts]: fixed.

**Comment by elamaran11** [comment]
> one minor comment. we need to think about how to maintain these versions across releases. For example, step 1 could be making an API call at synth time and checking wether the version used is indeed the compatible/default version for the target EKS version.

Sounds look a good idea, lets discuss about this.

**Review by shapirov103 — COMMENTED** [review]
Preliminary feedback

File: lib/addons/adot/index.ts [hunk 2]
```diff
@@ -6,6 +6,13 @@ import { CertManagerAddOn } from "../cert-manager";
 import { CoreAddOn, CoreAddOnProps } from "../core-addon";
 import { getAdotCollectorPolicyDocument } from "./iam-policy";
 import { semverComparator } from "../helm-addon/helm-version-checker";
+import { KubernetesVersion } from "aws-cdk-lib/aws-eks";
+
+const versionMap: Map<KubernetesVersion, string> = new Map([
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/adot/index.ts:11]: Yes its the same version as default for all 3 K8s versions.

File: lib/addons/core-addon/index.ts [hunk 3]
```diff
@@ -138,10 +139,43 @@ export class CoreAddOn implements ClusterAddOn {
...
+            {
+                const defaultVersions = response.addons?.flatMap(addon =>
+                    addon.addonVersions?.filter(version =>
+                      version.compatibilities?.some(compatibility => compatibility.defaultVersion === true)
+                    )
+                );
+
+                const version: string | undefined = defaultVersions[0]?.addonVersion;
+                if (!version) { 
+                    throw new Error("No default version found");
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/core-addon/index.ts]: Done.

File: lib/addons/core-addon/index.ts [hunk 4]
```diff
@@ -138,10 +139,43 @@ export class CoreAddOn implements ClusterAddOn {
...
+            }
+            else {
+                throw new Error("No add-on versions found");
+            }
+        }
+        catch (error) {
+            logger.warn(error);
+            logger.warn("Failed to retrieve add-on versions from EKS. Falling back to default version.");
+            if (!versionMap) {
+                throw new Error("No version map provided and no default version found");
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/core-addon/index.ts]: done

File: lib/addons/core-addon/index.ts [hunk 5]
```diff
@@ -138,10 +139,43 @@ export class CoreAddOn implements ClusterAddOn {
...
+                userLog.debug(`Core add-on ${this.coreAddOnProps.addOnName} has autoselected version ${version}`);
+                return version;
+            }
+            else {
+                throw new Error("No add-on versions found");
+            }
+        }
+        catch (error) {
+            logger.warn(error);
+            logger.warn("Failed to retrieve add-on versions from EKS. Falling back to default version.");
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/core-addon/index.ts]: done

File: test/coredns.test.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '../lib';
+import { KubernetesVersion } from 'aws-cdk-lib/aws-eks';
+
+test("Coredns Addon deploying correct default version of Addon for 1.28", async () => {
+    const app = new cdk.App();
+
+    await blueprints.EksBlueprint.builder()
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply test/coredns.test.ts:8]: Appreciate some samples on what is desired.

**Comment by elamaran11** [comment]
> Preliminary feedback

Fixed all comments, Please send a commit when you get a chance for jest fixes, we can make changes.

**Review by shapirov103 — APPROVED** [review]
LGTM, great work!

**Pull request merged**